SWIG/fdm.i

```diff
@@ -0,0 +1,2001 @@
+            
+        QL_ENSURE(pyResult != NULL,
+                  "failed to call setTime() on Python object");
+                                    
+        Py_XDECREF(pyResult);
+    }
+
+    Disposable<Array> apply(const Array& r) const {
+        return apply(r, "apply");        
+    }
+
+    Disposable<Array> apply_mixed(const Array& r) const {
+        return apply(r, "apply_mixed");        
+    }
+    
+    Disposable<Array> apply_direction(Size direction, const Array& r) const {
+        PyObject* pyArray = SWIG_NewPointerObj(
+            SWIG_as_voidptr(&r), SWIGTYPE_p_Array, 0);
+            
+        PyObject* pyResult 
+            = PyObject_CallMethod(callback_, "apply_direction", "kO", 
+                (unsigned long)(direction), pyArray);
+            
+        Py_XDECREF(pyArray); 
+            
+        return extractArray(pyResult, "apply_direction");        
+    }
+    
+    Disposable<Array> solve_splitting(
+        Size direction, const Array& r, Real s) const {
+
+        PyObject* pyArray = SWIG_NewPointerObj(
+            SWIG_as_voidptr(&r), SWIGTYPE_p_Array, 0);
+            
+        PyObject* pyResult 
+            = PyObject_CallMethod(callback_, "solve_splitting", "kOd", 
+                (unsigned long)(direction), pyArray, s);
+            
+        Py_XDECREF(pyArray); 
+            
+        return extractArray(pyResult, "solve_splitting");        
+    }
+    
+    Disposable<Array> preconditioner(const Array& r, Real s) const {
+        PyObject* pyArray = SWIG_NewPointerObj(
+            SWIG_as_voidptr(&r), SWIGTYPE_p_Array, 0);
+            
+        PyObject* pyResult 
+            = PyObject_CallMethod(callback_, "preconditioner", "Od",pyArray, s);
+            
+        Py_XDECREF(pyArray); 
+            
+        return extractArray(pyResult, "preconditioner");        
+    }
+
+  private:
+    Disposable<Array> apply(
+        const Array& r, const std::string& methodName) const {
+
+        PyObject* pyArray = SWIG_NewPointerObj(
+            SWIG_as_voidptr(&r), SWIGTYPE_p_Array, 0);
+            
+#if !defined(PY_VERSION_HEX) || PY_VERSION_HEX < 0x03040000         
+        std::vector<char> cstr(
+            methodName.c_str(), methodName.c_str() + methodName.size() + 1);  
+        PyObject* pyResult 
+            = PyObject_CallMethod(callback_, &cstr[0], "O", pyArray);
+#else
+        PyObject* pyResult 
+            = PyObject_CallMethod(callback_, methodName.c_str(), "O", pyArray);
+#endif            
+        Py_XDECREF(pyArray); 
+        
+        return extractArray(pyResult, methodName);        
+    }
+
+  private:        
+    PyObject* callback_;    
+};
+%}
+
+%shared_ptr(FdmLinearOpCompositeProxy)
+class FdmLinearOpCompositeProxy : public FdmLinearOpComposite {
+  public:
+    FdmLinearOpCompositeProxy(PyObject* callback);
+    
+    Size size() const;
+    void setTime(Time t1, Time t2);
+    
+    %extend {
```

Comment:
These (and the two above) are already declared as virtual in the base class.  I don't think you need to redeclare them here.  This should also remove the need for the `DeclareOperator` macro below and its `__COMMA` trick, since operators will only need to declare their constructor.

Reply:
The same goes for other hierarchies, e.g., boundary conditions below.